### ISSUE TITLE
Model File Export

### ISSUE THREAD

**Issue by CorneliusTantius** [IndoNLP/indonlu#29]
hi, I would like to ask. suppose I have done fine-tuning my model. how can I export it to physical app so that I can later load it to directly do prediction?

Thanks.

**Comment by SamuelCahyawijaya** [comment]
Hi @CorneliusTantius, can you explain further what do you mean by physical app? 

If what you mean is for mobile app or web app on the client side,  you can convert the model with ONNX into TensorFlow lite or TensorFlow.js. But due to the computation cost, it might not be able to work in real-time (you can except delay for processing). In this case, I would suggest if you want to create an app, you can implement the model as an API in your backend server (or another service that is called by your back end). 

If you make a service with our model in PyTorch, you can wrap the function into API with simple web server in Python such as Flask. When implementing the inference function, you should call `torch.eval()` to ignore all dropout layers and wrap the forward function with `torch.no_grad()` to prevent the model to store the activation during the forward function. You can follow the evaluation phase on the following example:
https://github.com/pytorch/examples/blob/cbb760d5e50a03df667cdc32a61f75ac28e11cbf/word_language_model/main.py#L138-L155

If you insist to deploy your model on-device (mobile app or client-side web app), there are other methods to shrink down the model size, such as distillation (https://arxiv.org/abs/2004.02984) and quantization (https://www.tensorflow.org/lite/performance/post_training_quantization). It would be a very useful resource if you try to develop a distill model for Indonesian language model (let say Indo-MobileBERT).

Let me know if you have any further questions regarding this issue. Thank you!

**Comment by CorneliusTantius** [comment]
Hi @SamuelCahyawijaya , the app I mean is some simple python app built using TKInter. That is why I wanted to kinda export the model(and what I expect is something like .pkl file I guess). 

FYI, I'm just following the steps in [Fine Tune Casa Example](https://github.com/indobenchmark/indonlu/blob/master/examples/finetune_casa.ipynb) but using different dataset. So my aim is something like exporting the `model` variable from the code and be able to load it once I want it.  

I think it would be fine if I'm the model size is huge. 

Also, I'm sorry for my bad explanation because I'm new to this. 

Thank you, sir!

**Comment by SamuelCahyawijaya** [comment]
I am not really familiar with TKInter, but if it is still using a python script, you can save and load the model parameters by using `torch.save()` and `torch.load()` functions.

For saving the model parameters, you can call `torch.save(model.state_dict(), "<MODEL_SAVE_PATH>")` after the fine-tuning phase.

For loading the model parameters, you can initialize the model first in you python app script and load the parameters with `model.load_state_dict(torch.load("<MODEL_SAVE_PATH>"))`.

**Comment by CorneliusTantius** [comment]
Ok sir. Thank you so much for your guidance, will try it ASAP. Once again, thankyou so much~
